### ISSUE TITLE
ng-serve JIT fails after update to v7 with ng-update for some apps

### ISSUE THREAD

**Issue by StephenFluin** [angular/angular-cli#12458]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
```

### Command (mark with an `x`)
```
- [x] serve
```

### Versions
This is part of the update, so coming from v6.x to v7.

```
Angular CLI: 6.2.4
Node: 10.11.0
OS: linux x64
Angular: 6.1.8
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, platform-server, router
... service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.8.3
@angular-devkit/build-angular     0.8.3
@angular-devkit/build-optimizer   0.8.3
@angular-devkit/build-webpack     0.8.3
@angular-devkit/core              0.8.3
@angular-devkit/schematics        0.8.3
@angular/cdk                      6.4.7
@angular/cli                      6.2.3
@angular/fire                     5.0.2
@angular/material                 6.4.7
@ngtools/webpack                  6.2.3
@schematics/angular               0.8.3
@schematics/update                0.8.3
rxjs                              6.3.2
typescript                        2.9.2
webpack                           4.19.1
```
to
```
Angular CLI: 7.0.0-rc.1
Node: 10.11.0
OS: linux x64
Angular: 7.0.0-rc.0
... animations, cdk, common, compiler, compiler-cli, core, forms
... http, language-service, material, platform-browser
... platform-browser-dynamic, platform-server, router
... service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.8.3
@angular-devkit/build-angular     0.8.3
@angular-devkit/build-optimizer   0.8.3
@angular-devkit/build-webpack     0.8.3
@angular-devkit/core              7.0.0-rc.1
@angular-devkit/schematics        7.0.0-rc.1
@angular/cli                      7.0.0-rc.1
@angular/fire                     5.0.2
@ngtools/webpack                  6.2.3
@schematics/angular               7.0.0-rc.1
@schematics/update                0.9.0-rc.1
rxjs                              6.3.3
typescript                        3.1.1
webpack                           4.19.1
```

### Repro steps
git clone git@github.com:GDGTC/devfest.git
cd devfest
yarn
ng update @angular/core @angular/cli @angular/material --next

ng serve  //fails at runtime 
ng serve --aot // works at runtime

### The log given by the failure
```
Uncaught Error: Can't resolve all parameters for ApplicationModule: (?).
    at syntaxError (compiler.js:1021)
    at CompileMetadataResolver.push../node_modules/@angular/compiler/fesm5/compiler.js.CompileMetadataResolver._getDependenciesMetadata (compiler.js:10943)
    at CompileMetadataResolver.push../node_modules/@angular/compiler/fesm5/compiler.js.CompileMetadataResolver._getTypeMetadata (compiler.js:10836)
    at CompileMetadataResolver.push../node_modules/@angular/compiler/fesm5/compiler.js.CompileMetadataResolver.getNgModuleMetadata (compiler.js:10704)
    at CompileMetadataResolver.push../node_modules/@angular/compiler/fesm5/compiler.js.CompileMetadataResolver.getNgModuleSummary (compiler.js:10514)
    at compiler.js:10628
    at Array.forEach (<anonymous>)
    at CompileMetadataResolver.push../node_modules/@angular/compiler/fesm5/compiler.js.CompileMetadataResolver.getNgModuleMetadata (compiler.js:10616)
    at CompileMetadataResolver.push../node_modules/@angular/compiler/fesm5/compiler.js.CompileMetadataResolver.getNgModuleSummary (compiler.js:10514)
    at compiler.js:10601```


### Desired functionality
Application functions at runtime without the `Can't resolve all parameters for ApplicationModule` error.


### Mention any other details that might be useful
<!-- Please include a link to the repo if this is related to an OSS project. -->

**Comment by clydin** [comment]
Looks like the migration script is not running.  `@angular-devkit/build-angular` should be at `0.9.0-rc.1`.

**Comment by StephenFluin** [comment]
@clydin Are you able to repro with my instructions? I'm hoping it's not another environmental issue specific to me.

**Comment by alan-agius4** [comment]
Hi @StephenFluin I have looked into this and indeed I managed to replicate the issue with the reproduction you provided.

The main problem here is caused because in your repo the  `@angular-devkit/build-angular` package is listed as a `dependency` and not `devDependency` https://github.com/GDGTC/devfest/blob/master/package.json#L16

In the migration schematic we are expecting that the later package is listed in `devDevendencies`. In fact you will end up with two `@angular-devkit/build-angular` on in your `devDevendencies` and another in the `dependencies`.

I have some ideas on how we can get around this. 

1) The `addPackageJsonDependency` method https://github.com/angular/angular-cli/blob/master/packages/schematics/angular/utility/dependencies.ts#L32  will check in `dev` and `dependencies` before trying adding it to the passed dependency type. So basically we will use the `dependency.type` only when it doesn't exist.
2) Do the checks solely for the migration script.
